addon/model.js

```diff
@@ -642,7 +642,16 @@ export default class MegamorphicModel extends EmberObject {
   // upon validation errors
   get errors() {
     if (this._errors === null) {
-      this._errors = Errors.create();
+      if (this._schema.useUnderlyingErrorsValue(this._modelName)) {
+        const errorsValue = this.unknownProperty('errors');
+        if (errorsValue) {
+          return errorsValue;
+        } else {
+          this._errors = Errors.create();
+        }
+      } else {
+        this._errors = Errors.create();
+      }
```

Comment:
if `useUnderlyingErrorsValue() === true` why do we fall back to the `DS.Errors` object at all?
2. related, I think this check should come before the `null` check. It doesn't make a huge difference, but if the user is opting-out of this check it's weird to avoid the normal caching that `this.unknownProperty` does.